posthog/api/insight.py

```diff
@@ -79,6 +81,18 @@ def create(self, validated_data: Dict, *args: Any, **kwargs: Any) -> DashboardIt
         else:
             raise serializers.ValidationError("Dashboard not found")
 
+    def update(self, instance: DashboardItem, validated_data: Dict) -> DashboardItem:
+        saved_pk = instance.pk
+        instance.pk = None
+        instance.id = None
+
+        new_instance = DashboardItem.objects.get(pk=saved_pk)
+
+        instance.parent = new_instance  # set cloned dashboard item's parent to the original dashboard item
```

Comment:
Is this function supposed to create a new DashboardItem? Also need to write a test for this.

Reply:
It creates a new DashboardItem with a parent id, and then it updates the original dashboard item as it's supposed to

Reply:
I might be late to the discussion (maybe you already discussed this when I was away) but not sure if this is the best way to do version history? Feels hacky. Seems like we could have a more abstract model that can handle historic version of stuff for any type of object, and handle this through model signals instead of here?